simapp/codec/codec.go

```diff
@@ -151,6 +153,33 @@ func (c *Codec) UnmarshalEvidenceJSON(bz []byte) (eviexported.Evidence, error) {
 	return evidence.GetEvidence(), nil
 }
 
+// MarshalProposal marshals a Proposal. It accepts a Proposal defined by the x/gov
+// module and uses the application-level Proposal type which has the concrete
+// Content implementation to serialize.
+func (c *Codec) MarshalProposal(p gov.Proposal) ([]byte, error) {
+	proposal := &Proposal{ProposalBase: p.ProposalBase}
+	if err := proposal.Content.SetContent(p.Content); err != nil {
+		return nil, err
+	}
+
+	return c.Marshaler.MarshalBinaryLengthPrefixed(proposal)
```

Comment:
```suggestion
	return c.Marshaler.MarshalBinaryBare(proposal)
```

Why should we be length-prefixing everywhere? It would only be needed for streaming messages AFAIK.

Reply:
We don't, but it's consistent.

Reply:
So you mean that we do it for every custom marshaler in this codec in one fell-swoop? That's fine. But we should do it when or before this goes into `codec/std` because then at that point it becomes state machine breaking for zones.

Reply:
I mean we do it for the entire SDK (where applicable). It doesn't matter if it's before or after the `std` move. This is all happening before the release (0.39) anyway.

Reply:
https://github.com/cosmos/cosmos-sdk/issues/5748